lua/entities/emplacements_turret_base/shared.lua

```diff
@@ -67,73 +98,72 @@ function ENT:Use( plr )
         if call == false then return end
 
         if IsValid( plr.CurrentEmplacement ) then
-            if SERVER then
-                -- plays sound on self to remind players this is an intended feature
-                self:EmitSound( "common/wpn_denyselect.wav", 60 )
-            end
-
-            return
+            plr.CurrentEmplacement:EmplacementDisconnect() -- hotswap emplacements! feels much better than being denied
+            
         end
-
-        self:SetShooter( plr )
-        self:StartShooting()
-        self.ShooterLast = plr
+        
+        self:EmplacementConnect( plr )
+        
     else
         if plr == self.Shooter then
-            self:SetShooter( nil )
-            self:FinishShooting()
+            self:EmplacementDisconnect()
         end
     end
 end
 
 function ENT:Think()
-    if not IsValid( self.turretBase ) and SERVER then
+    if not IsValid( self.turretBase ) then
+        if not SERVER then return end
         SafeRemoveEntity( self )
     else
```

Comment:
Something like this would let you get rid of the `else` and dedent everything inside.

It would probably be best to remove the need for this logic at all.

https://wiki.facepunch.com/gmod/Entity:CallOnRemove
```suggestion
    if not IsValid( self.turretBase ) then
        if SERVER then SafeRemoveEntity( self ) end
        return nil
    else
```